source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOVisibilityManager.mm

```diff
@@ -7,112 +7,212 @@
+
+    // setting hidden view to hidden again is a programming error
+    // as it requires to have equal or more times of the opposite value to be set
+    // in order to reverse it
+    if (!viewToBeHidden.isHidden) {
+        viewToBeHidden.hidden = YES;
+        // decrease the intrinsic content size by the intrinsic content size of
+        // `viewToBeHidden` otherwise, viewTobeHidden's size will be included
+        [hostView decreaseIntrinsicContentSize:viewToBeHidden];
+        [self changeVisiblityOfAssociatedViews:viewToBeHidden visibilityValue:YES contentStackView:hostView];
+    }
+
+    // if `viewToBeHidden` is a head, get new head if any, and hide its separator
+    if (isHead) {
+        NSUInteger headIndex = [self getHeadIndexOfVisibleViews];
+        if (headIndex != NSNotFound && headIndex < subviews.count) {
+            UIView *head = subviews[headIndex];
+            [self changeVisiblityOfSeparator:head visibilityValue:YES contentStackView:hostView];
         }
-        if (separator && separator.isHidden) {
-            separator.hidden = NO;
+    }
+}
+
+/// unhide `viewToBeUnhidden`. `hostView` is a superview of type ColumnView or ColumnSetView
+- (void)unhideView:(UIView *)viewToBeUnhidden hostView:(ACRContentStackView *)hostView
+{
+    if (!hostView || !viewToBeUnhidden || !_fillerSpaceManager) {
+        return;
+    }
+
+    NSArray<UIView *> *subviews = [hostView getContentStackSubviews];
+    if (!subviews) {
+        return;
+    }
+
+    NSUInteger index = [subviews indexOfObject:viewToBeUnhidden];
+    if (index == NSNotFound) {
+        return;
+    }
+
+    NSUInteger headIndex = [self getHeadIndexOfVisibleViews];
+    [self addVisibleView:index];
+    // check if the unhidden view will become a head
+    if ([self amIHead:index]) {
+        // only enable filler view associated with the `viewTobeUnhidden`
+        [self changeVisibilityOfPadding:viewToBeUnhidden visibilityValue:NO];
+        if (viewToBeUnhidden.isHidden) {
+            viewToBeUnhidden.hidden = NO;
+            [hostView increaseIntrinsicContentSize:viewToBeUnhidden];
         }
 
-        if (isLastView && self.padding && self.padding.isHidden) {
-            self.padding.hidden = NO;
+        // previous head view's separator becomes visible
+        if (headIndex != NSNotFound && headIndex < subviews.count) {
+            UIView *prevHeadView = subviews[headIndex];
+            [self changeVisiblityOfSeparator:prevHeadView visibilityValue:NO contentStackView:hostView];
         }
+    } else {
+        if (viewToBeUnhidden.isHidden) {
+            viewToBeUnhidden.hidden = NO;
+            [hostView increaseIntrinsicContentSize:viewToBeUnhidden];
+        }
+        [self changeVisiblityOfAssociatedViews:viewToBeUnhidden visibilityValue:NO contentStackView:hostView];
     }
 }
 
+- (void)updatePaddingVisibility
+{
+    [_fillerSpaceManager deActivateConstraintsForPadding];
+    [_fillerSpaceManager activateConstraintsForPadding];
```

Comment:
why is this happening? is it a rest or a refresh of some kind?

Reply:
visibility & padding changes are not done by the filler space manager. fillersapcemanger deactivates the old constraints. and activate new constraints changes from the visibility manager.